falcon/media/handlers.py

```diff
@@ -34,7 +35,7 @@ def _raise(self, *args, **kwargs):
     serialize = deserialize = _raise
 
 
-class Handlers(UserDict):
+class Handlers(MediaHandlers, UserDict):
```

Comment:
Maybe it's just me, but I don't see much benefit in adding this diamond inheritance here just to be able to type a private (underscored) method. Maybe we could just mark `self._resolve` as `Callable` or smth like that?

Reply:
IIRC they are stand in for protocols that aren't available on of the python version we target.
But maybe we can just use them in an type checking block, without needing to use the typing extension package

Reply:
Yes I meant them as a replacement of protocols.

Reply:
I think if we define them like this
```py
if TYPE_CHECKING:
  from typing import Protocol
  
  class MediaHandlers(Protocol): ...
```
then we use them like this
```py
if TYPE_CHECKING:
  from falcon.typing import MediaHandlers

  def i_get_an_handler(foo: MediaHandlers): ...
```

everything should work at runtime. It will fail type checking on python 3.7, but anyone who cares about type checking is most likely on a more recent version.

NOTE that we cannot use these as bases, so we cannot do `class Handlers(MediaHandlers, UserDict):`, but that should be ok in general, since the point of protocols is mostly to avoid having them as bases.

@vytas7 would that work for you?

Reply:
I've implemented it in accordance to @CaselIT . Please check if it's fine.

Reply:
We are probably going to drop active support for Python 3.7 in 4.0 anyway, so it doesn't matter if typing breaks there. I was just concerned with extra complexity and impact on performance (although it isn't the hottest path but still).

Just not "an handler" but "a handler" @CaselIT :slightly_smiling_face:

Reply:
I'm hopeless regarding use of an instead of a. I also routinely use it in fort of user :/